### PR TITLE
finally the baby yodas can be killed

### LINKED ISSUE #94: Make enemies lose hp and die when hit.

### PR DESCRIPTION
**Pull request by aggeagge**

### CODE REVIEW

File: src/common/simulation.ts [hunk 1]
```diff
@@ -101,12 +101,7 @@ export default abstract class Simulation {
   // despawns with a weird criteria atm, but is easily changed
   private despawnEnemies(): void {
     for (const enemy of Object.values(this.state.enemies)) {
-      if (
-        enemy.position.x < 0 ||
-        enemy.position.x > 400 ||
-        enemy.position.y < 0 ||
-        enemy.position.y > 400
-      ) {
+      if (enemy.health === 0) {
```
Comment (erikvader) [inline src/common/simulation.ts]:
```suggestion
      if (!enemy.alive) {
```

File: src/common/simulation.ts [hunk 2]
```diff
@@ -245,6 +244,20 @@ export default abstract class Simulation {
...
+    fixture: Fixture,
+    point: Vec2,
+    normal: Vec2,
+    fraction: number,
+  ): number {
+    const userData = fixture.getBody().getUserData() as { id: number }; ///to get id of the target
+    if (userData == null) {
+      return fraction;
+    }
+    this._state.enemies[userData.id].takeDamage(1);
```
Comment (erikvader) [inline src/common/simulation.ts]: https://en.wikipedia.org/wiki/Magic_number_(programming)

File: src/common/simulation.ts [hunk 3]
```diff
@@ -92,7 +92,7 @@ export default abstract class Simulation {
 
     const position = Vec2(48, 48);
 
-    const enemy = new Enemy(this._enemyIdCounter, 10, position);
+    const enemy = new Enemy(this._enemyIdCounter, 1, position);
```
Comment (erikvader) [inline src/common/simulation.ts:94]: https://en.wikipedia.org/wiki/Magic_number_(programming)

**Review by erikvader — CHANGES_REQUESTED** [review]

**Review by ponterik — APPROVED** [review]

**Review by stan3984 — APPROVED** [review]

File: src/common/simulation.ts [hunk 4]
```diff
@@ -245,6 +244,21 @@ export default abstract class Simulation {
...
+    fixture: Fixture,
+    point: Vec2,
+    normal: Vec2,
+    fraction: number,
+  ): number {
+    const userData = fixture.getBody().getUserData() as { id: number }; ///to get id of the target
+    if (userData == null || this._state.players[userData.id] !== undefined) {
+      return fraction;
+    }
+    console.log(userData.id);
```
Comment (Dojlando) [inline src/common/simulation.ts]: ta bort `console.log`

**Review by Dojlando — CHANGES_REQUESTED** [review]

**Review by erikvader — APPROVED** [review]

**Pull request merged**